src/sentry/sentry_metrics/querying/visitors/query_expression.py

```diff
@@ -422,6 +424,10 @@ def _extract_unit(self, timeseries: Timeseries) -> str | None:
 
         parsed_mri = parse_mri(timeseries.metric.mri)
         if parsed_mri is not None:
+            if parsed_mri.name.startswith(SPAN_ATTRIBUTE_PREFIX):
+                rule_id = int(parsed_mri.name[len(SPAN_ATTRIBUTE_PREFIX) :])
```

Comment:
What if we extend the `ParsedMRI` to have a method which tries to get the rule id? Maybe in a subclass or just as standalone method, I feel like we might need this more than once. This method could also embed good error handling in case of failure of `int` conversion.

Reply:
Added a property to the class to do this